City-of-Helsinki/drupal-helfi-rekry
# Uhf 7763 elastic furthering

# [UHF-7763](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7763)
Combined resolution for:
* https://helsinkisolutionoffice.atlassian.net/browse/UHF-7763
* https://helsinkisolutionoffice.atlassian.net/browse/UHF-7870

## What was done
* Resolve issues with job importing
  * Timezone issues
  * Old translations not unpublishing correctly
* Add checkbox filters to search
* Enable searching by recruitment id
* Add finnish analyzer to search
* Fix some translations
## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout origin/UHF-7763-elastic-furthering`
  * `make fresh` **OR** `make build; make drush-cim; make drush-updb; make drush-locale-update`
* Run `make drush-cr`

## How to test

First, run migrations and all subsequent tasks (this might take a while):
```
drush mim helfi_rekry_jobs:all --update; drush mim helfi_rekry_jobs:all_sv --update; drush mim helfi_rekry_jobs:all_en --update; drush queue:run helfi_rekry_job_translations; drush cron; drush sapi-rt; drush sapi-c; drush sapi-i
```

Check that finnish analyzer is in place: GET `http://localhost:9201/job_listings/_settings` and you should see
```
                "analysis": {
                    "analyzer": {
                        "default": {
                            "type": "finnish"
                        }
                    }
                },
```
within the response.

Make a landing page with job_search paragraph in it (or go to https://helfi-rekry.docker.so/fi/avoimet-tyopaikat) if you're using db from testing.

Test the search app: 
* Search with this string: sote-03-83-23. You should get some results.
* You should see checkboxes under the other search filters. Checking them (only them) should produce some results.
* You should not be seeing any mistranslations on any languages



[UHF-7763]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-7763?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
